### PR TITLE
feat: support prometheus new client of micrometer 1.13.0 version

### PR DESCRIPTION
**Pull request by qyi71**
## What is the purpose of the change
support prometheus new client of micrometer 1.13.0 version
 ref: #14228 
@AlbumenJ 
@songxiaosheng 

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by songxiaosheng** [comment]
@AlbumenJ help check sample ci,thanks

File: dubbo-metrics/dubbo-metrics-prometheus/src/main/java/org/apache/dubbo/metrics/prometheus/PrometheusMetricsReporter.java [hunk 1]
```diff
@@ -93,7 +96,7 @@ private void schedulePushJob() {
     protected void push(PushGateway pushGateway, String job) {
         try {
             resetIfSamplesChanged();
-            pushGateway.pushAdd(prometheusRegistry.getPrometheusRegistry(), job);
+            pushGateway.push();
```
Comment (songxiaosheng) [inline dubbo-metrics/dubbo-metrics-prometheus/src/main/java/org/apache/dubbo/metrics/prometheus/PrometheusMetricsReporter.java:99]: Is there a document available

**Comment by qyi71** [comment]
@songxiaosheng  please help look this , maybe is not my fault

File: dubbo-dependencies-bom/pom.xml [hunk 2]
```diff
@@ -717,15 +717,14 @@
...
       <dependency>
         <groupId>io.prometheus</groupId>
-        <artifactId>simpleclient</artifactId>
+        <artifactId>prometheus-metrics-core</artifactId>
         <version>${prometheus_client.version}</version>
       </dependency>
       <dependency>
         <groupId>io.prometheus</groupId>
-        <artifactId>simpleclient_pushgateway</artifactId>
+        <artifactId>prometheus-metrics-exporter-pushgateway</artifactId>
```
Comment (AlbumenJ) [inline dubbo-dependencies-bom/pom.xml:725]: We can only make this breaking change in 3.3

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Comment by qyi71** [comment]
@AlbumenJ  PTAL

**Comment by AlbumenJ** [comment]
image

**Pull request closed without merging**